Python/fhirabstractbase.py

```
@@ -155,21 +169,36 @@ def as_json(self):
                     .format(nonop, self))
         return js
 
-    def _cast(self, value, typ):
-        if type(value) == typ:
-            return value
-        if typ == bool:
-            if isinstance(value, str):
-                if value.lower() in ['0', 'no', 'zero', 'false', 'off', 'disable']:
-                    return False
-                elif value.lower() in ['1', 'yes', 'one', 'true', 'on', 'enable']:
-                    return True
-                else:
-                    return None
-        try:
-            return typ(value)
-        except TypeError:
+    def _cast(self, value, typ, is_list=False):
+        if value is None:
             return None
+        if is_list:
+            if not isinstance(value, list):
+                return None
+            result = []
+            for item in value:
+                result.append(self._cast(item, typ))
+            return result
+        else:
+            if type(value) == typ:
+                return value
+            if typ == bool:
+                if isinstance(value, str):
+                    if value.lower() in ['0', 'no', 'zero', 'false', 'off', 'disable']:
+                        return False
+                    elif value.lower() in ['1', 'yes', 'one', 'true', 'on', 'enable']:
+                        return True
+                    else:
+                        return None
+            try:
+                return typ(value)
+            except (TypeError, ValueError):
```

If the goal is to silently skip the exception better catch `Exception` itself. Maybe log it too.